duckduckgo/Android
# Improve connectivity issues

Task/Issue URL: https://app.asana.com/0/0/1202075724102060/f

### Description
To improve connectivity issues this PR will track network hand-offs and reconfigure the VPN DNS accordingly

### Steps to test this PR

_Manual test_
- [x] install from this branch and enable AppTP
- [x] go to settings -> AppTP dev settings
- [x] verify only `IPv6 support` option is ON in the `CONFIG` section
- [x] verify apps work and AppTP blocks trackers both in WIFI and CELLULAR networks
- [x] go to settings => AppTP dev settings and enable `Network Switch Handling`
- [x] (VPN should automatically restart)
- [x] Ensure you are on WIFI
- [x] Using Android settings perform a network switch WIFI to CELLULAR
- [x] Verify the VPN restarts and that AppTP blocks trackers and apps are able to load content
- [x] Using Android settings perform a network switch CELLULAR to WIFI
- [x] Verify the VPN restarts and that AppTP blocks trackers and apps are able to load content
- [x] go to settings => AppTP dev settings and enable `Set active network DNS`
- [x] (VPN should automatically restart)
- [x] verify apps work and AppTP blocks trackers both in WIFI and CELLULAR networks


NOTE: The feature `always set DNS` is added to internal features but we don't plan to use it for now in production, see [this task](https://app.asana.com/0/69071770703008/1202075724102015/f) for more info about why
